src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -6933,12 +7011,568 @@ public static void Main(string[] args)
+                Diagnostic(ErrorCode.ERR_DefaultValueBeforeRequiredValue, ")").WithLocation(5, 44));
+        }
+
+        [Fact]
+        public void LambdaWithDefaultBeforeRequired2()
+        {
+            var source =
+@"class Program
+{
+    public static void Main()
+    {
+        var lam = (int x, int y = 3, int z) => x + y + z;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,43): error CS1737: Optional parameters must appear after all required parameters
+                //         var lam = (int x, int y = 3, int z) => x + y + z;
+                Diagnostic(ErrorCode.ERR_DefaultValueBeforeRequiredValue, ")").WithLocation(5, 43));
+        }
+
+
+        [Fact]
+        public void AnonymousMethodDefaultBeforeRequired2()
+        {
+            var source =
+@"class Program
+{
+    public static void Main()
+    {
+        var lam = delegate(int x, int y = 3, int z) { return x + y + z; };
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                    // (5,51): error CS1737: Optional parameters must appear after all required parameters
+                    //         var lam = delegate(int x, int y = 3, int z) { return x + y + z; };
+                    Diagnostic(ErrorCode.ERR_DefaultValueBeforeRequiredValue, ")").WithLocation(5, 51));
+        }
+
+        [Fact]
+        public void LambdaWithDefaultTypeMismatchLiteral()
+        {
+            var source = """
+class Program
+{
+    public static void Main()
+    {
+        var lam = (int x = "abcdef") => x;
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                    // (5,24): error CS1750: A value of type 'string' cannot be used as a default parameter because there are no standard conversions to type 'int'
+                    //         var lam = (int x = "abcdef") => x;
+                    Diagnostic(ErrorCode.ERR_NoConversionForDefaultParam, "x").WithArguments("string", "int").WithLocation(5, 24));
+        }
+
+        [Fact]
+        public void AnonymousMethodWithDefaultTypeMismatchLiteral()
+        {
+            var source = """
+class Program
+{
+    public static void Main()
+    {
+        var lam = delegate(int x = "abcdef") { return x; };
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,32): error CS1750: A value of type 'string' cannot be used as a default parameter because there are no standard conversions to type 'int'
+                //         var lam = delegate(int x = "abcdef") { return x; };
+                Diagnostic(ErrorCode.ERR_NoConversionForDefaultParam, "x").WithArguments("string", "int").WithLocation(5, 32));
+        }
+
+        [Fact]
+        public void LambdaWithDefaultTypeMismatchConstantExpression()
```

Comment:
>LambdaWithDefaultTypeMismatchConstantExpression [](http://example.com/codeflow?start=20&length=47)

The type mismatch cases above seem sufficient. These two can probably be removed.